citesphere-model/src/main/java/edu/asu/diging/citesphere/data/bib/impl/CitationMongoDao.java

```diff
@@ -55,18 +55,29 @@ public List<? extends ICitation> findCitations(String groupId, long start, int p
         query.limit(pageSize);
         return mongoTemplate.find(query, Citation.class);
     }
-
+    
     @Override
-    public List<? extends ICitation> findCitationsByContributorUri(List<String> groupIds, long start, int pageSize, String uri) {
-        Query query = new Query();
-        query.addCriteria(Criteria.where("group").in(groupIds));
+    public Query buildQueryToFindCitationsByContributorUri(List<String> groupIds, String uri) {
```

Comment:
this should not need to be `public` and also, let's not be overly wordy, how about `buildCitationsByContributorQuery`?

Reply:
and then make `uri` -> `contributorUri` and add javadoc